Provide Stack support?

I have experimented with using HMock with Stack, and I get the following error when using GHC 8.6 and 8.8:

```
/tmp/stack-b8726112c6310f6a/HMock-0.5.0.0/src/Test/HMock/Internal/State.hs:202:7: error:
    • Couldn't match type ‘MockT m’ with ‘ReaderT (MockState m) m’
        arising from the coercion of the method ‘unliftio-core-0.1.2.0:Control.Monad.IO.Unlift.askUnliftIO’
          from type ‘ReaderT
                       (MockState m)
                       m
                       (unliftio-core-0.1.2.0:Control.Monad.IO.Unlift.UnliftIO
                          (ReaderT (MockState m) m))’
            to type ‘MockT
                       m
                       (unliftio-core-0.1.2.0:Control.Monad.IO.Unlift.UnliftIO (MockT m))’
    • When deriving the instance for (MonadUnliftIO (MockT m))
    |
202 |       MonadUnliftIO
    |       ^^^^^^^^^^^^^
```

I do not know why I get this error when using Stack but not Cabal, as I use the same version of relevant packages in both cases.  Do you know what might be causing this?

I was able to get around this error by writing the instance for `MonadUnliftIO` manually instead of deriving it:

```haskell
instance MonadUnliftIO m => MonadUnliftIO (MockT m) where
  withRunInIO inner = MockT $ withRunInIO $ \run -> inner (run . unMockT)
```

If you would like to include this change for better Stack support, I am happy to create a pull request.

----

The following are my Stack configurations for a project that uses HMock.  (For the HMock package itself, the `HMock-0.5.0.0` dependency would need to be removed and the `doctest` dependencies would need to be added.  I can provide these files if wanted.)

GHC 8.6.5:

```yaml
resolver: lts-14.27

packages:
  - .

extra-deps:
  - binary-0.8.6.0
  - constraints-0.13
  - containers-0.6.5.1
  - explainable-predicates-0.1.2.0
  - extra-1.7.10
  - HMock-0.5.0.0
  - regex-base-0.94.0.1
  - regex-tdfa-1.3.1.1
  - syb-0.7.2.1
  - text-1.2.3.1
  - type-equality-1
  - unliftio-0.2.20
```

GHC 8.8.4:

```yaml
resolver: lts-16.31

packages:
  - .

extra-deps:
  - constraints-0.13
  - explainable-predicates-0.1.2.0
  - HMock-0.5.0.0
  - syb-0.7.2.1
  - unliftio-0.2.20
```

GHC 8.10.7:

```yaml
resolver: lts-18.12

packages:
  - .

extra-deps:
  - explainable-predicates-0.1.2.0
  - HMock-0.5.0.0
```

GHC 9.0.1:

```yaml
resolver: nightly-2021-10-01

packages:
  - .
```

(Update: `HMock` and `explainable-predicates` are included in `nightly-2021-10-01`! :+1: )

--- a/src/Test/HMock/Internal/State.hs
+++ b/src/Test/HMock/Internal/State.hs
@@ -36,7 +36,7 @@ import Test.HMock.Internal.Util (Located)
 import Test.HMock.Mockable (Mockable (..))
 import UnliftIO
   ( MonadIO,
-    MonadUnliftIO,
+    MonadUnliftIO(withRunInIO),
     STM,
     TVar,
     atomically,
@@ -198,12 +198,16 @@ newtype MockT m a where
       MonadBase b,
       MonadCatch,
       MonadMask,
-      MonadThrow,
-      MonadUnliftIO
+      MonadThrow
     )
 
 instance MonadTrans MockT where
   lift = MockT . lift
+
+-- Note: The 'MonadUnliftIO' instance is implemented manually because deriving
+-- it causes compilation failure in GHC 8.6 and 8.8.  (See issue #23.)
+instance MonadUnliftIO m => MonadUnliftIO (MockT m) where
+  withRunInIO inner = MockT $ withRunInIO $ \run -> inner (run . unMockT)
 
 -- | Applies a function to the base monad of 'MockT'.
 mapMockT :: (m a -> m b) -> MockT m a -> MockT m b

